src/QueryBuilder/QueryViewer/QueryViewer.js

```diff
@@ -0,0 +1,124 @@
+import React, { memo, useEffect, useMemo } from 'react';
+import PropTypes from 'prop-types';
+import { Col, Row, Accordion, AccordionSet, MultiColumnList, Headline } from '@folio/stripes/components';
+import { PrevNextPagination, usePagination } from '@folio/stripes-acq-components';
+import QueryLoader from '../QueryLoader/QueryLoader';
+import { getTableMetadata } from '../helpers';
+
+const QueryViewer = ({
+  showPagination = true,
+  contentData = [],
+  limit = 100,
+  offset = 0,
+  onPageChange,
+  entityType,
+  headline,
+  headlineEnd,
+  visibleColumns,
+  onSetDefaultVisibleColumns,
+  onSetDefaultColumns,
+  accordionHeadline,
+  isInProgress,
+  inProgressTitle,
+  height,
+}) => {
+  const { changePage, pagination } = usePagination({ limit, offset });
```

Comment:
as I remember this hook works with location, how will you support it in modal and on the page at the same time?

Reply:
`  useEffect(() => { onPageChange?.(pagination); })` this part allow us work with it without location dependencies, call the parent handler on each paging change

Reply:
but usePagination works with location, it's design issue

Reply:
maybe you know alternative which we can use from folio libraries? Or we can just add configuration in usePagination about disabling changing of search params.

Reply:
discussed offline